src/app/space/create/deployments/apps/deployments-apps.component.html

```diff
@@ -1,32 +1,82 @@
 <deployments-toolbar (onFilterChange)="filterChange($event)" (onSortChange)="sortChange($event)" [resultsCount]="resultsCount">
 </deployments-toolbar>
 
-<div class="cards-pf">
-  <div class="row row-cards-pf">
-    <div class="col-xs-12 col-sm-4 col-md-2">
-      <h3>
-        <b>Applications</b>
-      </h3>
-    </div>
-    <div class="col-xs-12 col-sm-4 col-md-2">
-      <h3>
-        <b>Environments</b>
-      </h3>
+<div *ngIf="(applications.first() | async); else loading">
```

Comment:
Much better, although it's still possible for the request for applications to resolve very quickly and the request for environments to resolve very slowly, in which case the loading screen is again prematurely hidden. Perhaps rather than displaying the loading screen conditionally on just `applications` this should be a merged observable of both `applications` and `environments`' first results?